src/main/java/uk/gov/hmcts/divorce/solicitor/event/SolicitorConfirmService.java

```diff
@@ -0,0 +1,108 @@
+
+@Slf4j
+@Component
+public class SolicitorConfirmService implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String SOLICITOR_CONFIRM_SERVICE = "solicitor-confirm-service";
+
+    @Autowired
+    private SolicitorSubmitConfirmService solicitorSubmitConfirmService;
+
+    private final List<CcdPageConfiguration> pages = List.of(
+        new SolConfirmService()
+    );
+
+    @Override
+    public void configure(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        final PageBuilder pageBuilder = addEventConfig(configBuilder);
+        pages.forEach(page -> page.addTo(pageBuilder));
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> aboutToSubmit(final CaseDetails<CaseData, State> details,
+                                                                       final CaseDetails<CaseData, State> beforeDetails) {
+        final CaseData caseData = details.getData();
+
+        log.info("Solicitor confirm service about to submit callback invoked with service method {}",
+            caseData.getApplication().getSolServiceMethod().toString());
+
+        final List<String> validationErrors = validateConfirmService(caseData.getApplication());
+
+        if (!validationErrors.isEmpty()) {
+            log.info("ConfirmService Validation errors: ");
+            for (String error : validationErrors) {
+                log.info(error);
+            }
+
+            return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+                .data(caseData)
+                .errors(validationErrors)
+                .state(details.getState())
+                .build();
+        }
+
+
+        final CaseDetails<CaseData, State> updateDetails = solicitorSubmitConfirmService.submitConfirmService(details);
+
+        log.info("Due date after submit Task is {}", updateDetails.getData().getDueDate());
+
+        return AboutToStartOrSubmitResponse.<CaseData, State>builder()
+            .data(updateDetails.getData())
+            .state(updateDetails.getState())
+            .build();
+    }
+
+    private PageBuilder addEventConfig(ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        return new PageBuilder(configBuilder
+            .event(SOLICITOR_CONFIRM_SERVICE)
+            .forStateTransition(AwaitingService, AwaitingAos)
+            .name("Solicitor Confirm Service")
+            .description("Solicitor Confirm Service")
+            .showSummary()
+            .aboutToSubmitCallback(this::aboutToSubmit)
+            .explicitGrants()
+            .grant(CREATE_READ_UPDATE, SOLICITOR)
+            .grant(READ,
+                CASE_WORKER,
+                SUPER_USER,
+                LEGAL_ADVISOR));
+    }
+
+    private List<String> validateConfirmService(Application application) {
+        List<String> errors = new ArrayList<>();
+
+        if (!application.getSolServiceMethod().equals(ServiceMethod.SOLICITOR_SERVICE)) {
```

Comment:
There is an Application method that checks for solicitorService method, isSolicitorServiceMethod

Reply:
Ah, will update